src/shell/Shell.cpp

```diff
@@ -332,17 +231,16 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
                     nullptr));
             }
         } else if (import->moduleName() == "wasi_snapshot_preview1") {
-            // TODO wasi
-            if (import->fieldName() == "proc_exit") {
-                auto ft = functionTypes[SpecTestFunctionTypes::I32];
+            Walrus::WASI::WasiFunc* wasiImportFunc = wasi->find(import->fieldName());
+            if (wasiImportFunc == nullptr) {
+                break;
```

Comment:
Is it okay to just stop (break) the whole imporing process when unimplemented wasi function is imported?
What if the remained import elements (`importTypes`) are valid?

Reply:
You are right, I honestly did not think about the unimplemented functions as I plan to implement all of preview1.

As far as I remember the documentation doesn't really mention this, I would believe it is left to the implementing environment. I would like your opinion on this matter. Which would you prefer to just communicate the error to the user and continue or should we leave it as it is now ( exit with error `Insufficient import` ) ? I am open to any other ideas too.

Reply:
How about handling unimplemented WASI features simply?
I mean that importing only supported WASI functions and just ignoring others.
```c++
if (wasiImportFunc  != nullptr) {
// import wasi function
...
}
// continue importing of next element
```

Reply:
I am okay with this and have updated the patch to implement it.